openhcl/hcl/src/mapped_page.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::fs::File;
+use std::io;
+use std::os::fd::AsRawFd;
+use std::ptr::NonNull;
+
+pub(crate) struct MappedPage<T>(NonNull<T>);
```

Comment:
Perhaps change this to `NonNull<UnsafeCell<T>>` and hand out real references to it instead of just pointers.